numpy/f2py/backends/meson_backend.py

```diff
@@ -0,0 +1,183 @@
+
+	def dependencies_substitution(self) -> None:
+		self.substitutions["dependencies_list"] = ", ".join([f"dependency('{dependecy}')" for dependecy in self.dependencies])
+		
+	def include_directories_subtitution(self) -> None:
+		self.substitutions["include_directories_list"] = ", ".join([f"include_directories('{include_directory}')" for include_directory in self.include_directories])
+
+	def global_flags_substitution(self) -> None:
+		fortran_compiler_flags = self.fortran_flags + self.optimization_flags + self.architecture_flags
+		c_compiler_flags = self.optimization_flags + self.architecture_flags
+		self.substitutions["fortran_global_args"] = fortran_compiler_flags
+		self.substitutions["c_global_args"] = c_compiler_flags
+	
+	def macros_substitution(self) -> None:
+		self.substitutions["macros"] = ""
+		if self.define_macros:
+			self.substitutions["macros"] = ",".join(f"\'-D{macro[0]}={macro[1]}\'" if macro[1] else f"-D{macro[0]}" for macro in self.define_macros)
+		if self.undef_macros:
+			self.substitutions["macros"] += "," + ",".join(f"\'-U{macro}\'" for macro in self.undef_macros)
+	
+	def linker_substitution(self) -> None:
+		self.substitutions["linker_args"] = ""
+		if self.linker_libpath:
+			linker_libpath_subs = ",".join(f"-L{libpath}" for libpath in self.linker_libpath)
+			self.substitutions["linker_args"] += linker_libpath_subs
+		if self.linker_libname:
+			linker_libname_subs = ",".join(f"-l{libname}" for libname in self.linker_libname)
+			self.substitutions["linker_args"] += f",{linker_libname_subs}"
+
+	def generate_meson_build(self) -> str:
+		for node in self.pipeline:
+			node()
+		template = Template(self.meson_build_template)
+		return template.substitute(self.substitutions)
+
+
+class MesonBackend(Backend):
+
+	def __init__(self, module_name: str = 'untitled', fortran_compiler: str = None, c_compiler: str = None, f77exec: Path = None, f90exec: Path = None, f77_flags: list[str] = None, f90_flags: list[str] = None, include_paths: list[Path] = None, include_dirs: list[Path] = None, external_resources: list[str] = None, linker_libpath: list[Path] = None, linker_libname: list[str] = None, define_macros: list[tuple[str, str]] = None, undef_macros: list[str] = None, debug: bool = False, opt_flags: list[str] = None, arch_flags: list[str] = None, no_opt: bool = False, no_arch: bool = False) -> None:
+		self.meson_build_dir = "builddir"
+		if f77_flags is None:
+			f77_flags = []
+		if include_paths is None:
+			include_paths = []
+		if include_dirs is None:
+			include_dirs = []
+		if external_resources is None:
+			external_resources = []
+		if linker_libpath is None:
+			linker_libpath = []
+		if linker_libname is None:
+			linker_libname = []
+		if define_macros is None:
+			define_macros = []
+		if undef_macros is None:
+			undef_macros = []
+		if f77_flags is None:
+			f77_flags = []
+		if f90_flags is None:
+			f90_flags = []
+		if opt_flags is None:
+			opt_flags = []
+		if arch_flags is None:
+			arch_flags = []
```

Comment:
These flags don't quite fit anymore (the difference between F77 and F90 is not needed, nor are the exact paths to the compiler; the `--opt`, `--arch` etc. are specified in a very non-standard way, Meson has a better way to do things like `debug` flags). Is there a plan to clean this up? Maybe they should be kept unchanged for now, and then when `numpy.distutils` is gone they can be updated?